media_driver/linux/common/codec/ddi/media_ddi_encode_jpeg.cpp

```diff
@@ -703,7 +715,7 @@ VAStatus DdiEncodeJpeg::QualityScaleQmatrix()
                 quantValue = 255;
             }
 
-            quantMatrix->m_quantTable[qMatrixCount].m_qm[i] = (uint16_t)quantValue;
+            quantMatrix->m_quantTable[qMatrixCount].m_qm[zigzag_order[i]] = (uint16_t)quantValue;
```

Comment:
Could you please clarify this a little? `m_qm` is modified in-place,
so, say, after iteration `i = 2` puts scaled value to `m_qm[8]`,  on iteration with `i = 8`
`quantValue` will be updated one, not original from an application. Is it intentional?

Reply:
hi @Alexandr-Konovalov , the matrix from application should be zigzag order, but HW need Raster order, I just convert it from ZigZag to Raster

Reply:
@XinfengZhang Thanks, I understand that. What I do not understand: is it OK to apply quality to the matrix in-place? Say, for `i = 8` the code puts to `quantMatrix->m_quantTable[qMatrixCount].m_qm[17]` quantValue that was scaled from output of iteration `i = 2`, not from value provided by an application.

Reply:
You can't convert order inplace. Because matrix is read in one order and is written in another.